cms/djangoapps/contentstore/views/tests/test_item.py

```diff
@@ -680,6 +681,152 @@ def verify_name(source_usage_key, parent_usage_key, expected_name, display_name=
+
+        vert2 = self.create_xblock(parent_usage_key=self.seq_usage_key, display_name='vertical2', category='vertical')
+        self.vert2_usage_key = self.response_usage_key(vert2)
+
+        # create problem and an html component
+        problem1 = self.create_xblock(parent_usage_key=self.vert_usage_key, display_name='problem1', category='problem')
+        self.problem_usage_key = self.response_usage_key(problem1)
+
+        html1 = self.create_xblock(parent_usage_key=self.vert_usage_key, display_name='html1', category='html')
+        self.html_usage_key = self.response_usage_key(html1)
+
+    def _move_component(self, source_usage_key, dest_usage_key, source_index=None):
+        """
+        Helper method to send move request and returns the response.
+
+        Arguments:
+            source_usage_key (BlockUsageLocator): Locator of source item.
+            dest_usage_key (BlockUsageLocator): Locator of destination item.
+            source_index (int): If provided, insert source item at the provided index location in dest_usage_key item.
+
+        Returns:
+            resp (JsonResponse): Response after the move operation is complete.
+        """
+        data = {
+            'move_source_locator': unicode(source_usage_key),
+            'dest_source_locator': unicode(dest_usage_key)
+        }
+        if source_index is not None:
+            data['source_index'] = source_index
+        resp = self.client.ajax_post(reverse('contentstore.views.xblock_handler'), json.dumps(data))
+        return resp
+
+    def assert_move_item(self, source_usage_key, dest_usage_key, source_index=None):
+        """
+        Assert move component.
+
+        Arguments:
+            source_usage_key (BlockUsageLocator): Locator of source item.
+            dest_usage_key (BlockUsageLocator): Locator of destination item.
+            source_index (int): If provided, insert source item at the provided index location in dest_usage_key item.
+        """
+        parent_loc = self.store.get_parent_location(source_usage_key)
+        actual_souce_index = _get_source_index(source_usage_key)
+        response = self._move_component(source_usage_key, dest_usage_key, source_index)
+        self.assertEqual(response.status_code, 200)
+        response = json.loads(response.content)
+        self.assertEqual(response['move_source_locator'], unicode(source_usage_key))
+        self.assertEqual(response['dest_source_locator'], unicode(dest_usage_key))
+        source_index = source_index if source_index is not None else actual_souce_index
+        self.assertEqual(response['source_index'], source_index)
+        new_parent_loc = self.store.get_parent_location(source_usage_key)
+        self.assertEqual(new_parent_loc, dest_usage_key)
+        self.assertNotEqual(parent_loc, new_parent_loc)
+
+    def test_move_component(self):
+        """
+        Test move component with different xblock types.
+        """
+        for source_usage_key, dest_usage_key in [
```

Comment:
I can not use DDT because I need class variables as in `self.html_usage_key` etc. which are not available outside the scope of the methods.